indico/MaKaC/webinterface/pages/error.py

```diff
@@ -14,17 +14,27 @@
 # You should have received a copy of the GNU General Public License
 # along with Indico; if not, see <http://www.gnu.org/licenses/>.
 
-from MaKaC.webinterface.wcomponents import WTemplated
+from flask import session
 
+from MaKaC.webinterface.pages.base import WPDecorated, WPJinjaMixin
+from MaKaC.accessControl import AccessWrapper
 
-class WErrorWSGI(WTemplated):
 
-    def __init__(self, ex):
-        WTemplated.__init__(self)
-        self._ex = ex
+class WErrorWSGI(WPDecorated, WPJinjaMixin):
 
-    def getVars(self):
-        vars = WTemplated.getVars(self)
-        vars["errorTitle"] = str(self._ex[0])
-        vars["errorText"] = str(self._ex[1])
-        return vars
+    def __init__(self, message, description):
+        WPDecorated.__init__(self, None)
+        self._message = message
+        self._description = description
+
+    def _getBody(self, params):
+        return self._getPageContent({
+            '_jinja_template': 'error.html',
+            'error_message': self._message,
+            'error_description': self._description})
+
+    def _getAW(self):
+        return AccessWrapper(session.user.as_avatar if session.user else None)
```

Comment:
is that used anywhere? otherwise let's use `session.user` (I think there's also a `session.avatar` property btw) directly